### PR TITLE
Moves media publisher info into separate file

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7126

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
covered with tests

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/database/database_media_publisher_info.cc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+#include "base/bind.h"
+#include "base/strings/stringprintf.h"
+#include "brave/components/brave_rewards/browser/database/database_media_publisher_info.h"
+#include "sql/statement.h"
+#include "sql/transaction.h"
+
+namespace brave_rewards {
+
+namespace {
+  const char* table_name_ = "media_publisher_info";
```
Comment (tmancey) [inline components/brave_rewards/browser/database/database_media_publisher_info.cc]: `table_name_` is not a member variable so no need for underscore at end of the name

Comment (tmancey) [reply components/brave_rewards/browser/database/database_media_publisher_info.cc]: nit: `const char table_name[] = "media_publisher_info";`

File: components/brave_rewards/browser/database/publisher_info_database.h [hunk 2]
```diff
@@ -200,6 +200,7 @@ class PublisherInfoDatabase {
   std::unique_ptr<DatabaseUnblindedToken> unblinded_token_;
   std::unique_ptr<DatabaseContributionInfo> contribution_info_;
   std::unique_ptr<DatabasePendingContribution> pending_contribution_;
+  std::unique_ptr<DatabaseMediaPublisherInfo> media_publisher_info;
```
Comment (tmancey) [inline components/brave_rewards/browser/database/publisher_info_database.h]: member variable should end with an underscore

File: components/brave_rewards/browser/database/publisher_info_database.cc [hunk 3]
```diff
@@ -359,6 +365,47 @@ bool PublisherInfoDatabase::RestorePublishers() {
...
+
+  bool initialized = Init();
+  DCHECK(initialized);
+
+  if (!initialized) {
+    return false;
+  }
+
+  // We will use every attribute from publisher_info
+  std::string query = "SELECT pi.publisher_id, spi.status, pi.name,"
```
Comment (tmancey) [inline components/brave_rewards/browser/database/publisher_info_database.cc]: nit: const

File: components/brave_rewards/browser/database/publisher_info_database.cc [hunk 4]
```diff
@@ -359,6 +365,47 @@ bool PublisherInfoDatabase::RestorePublishers() {
   return restore_q.Run();
 }
 
+bool PublisherInfoDatabase::GetExcludedList(
+    ledger::PublisherInfoList* list) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  CHECK(list);
```
Comment (tmancey) [inline components/brave_rewards/browser/database/publisher_info_database.cc]: CHECK will throw an exception on release builds. Recommend to change to:```DCHECK(list);
if (!list) {
  return false;
}```

File: components/brave_rewards/browser/database/database_media_publisher_info.h [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  bool CreateIndex(sql::Database* db) override;
+
+  bool InsertOrUpdate(
+      sql::Database* db,
+      const std::string& media_key,
+      const std::string& publisher_key);
+
+  ledger::PublisherInfoPtr GetRecord(
+    sql::Database* db,
```
Comment (tmancey) [inline components/brave_rewards/browser/database/database_media_publisher_info.h]: nit: 4 spaces

File: components/brave_rewards/browser/database/publisher_info_database.h (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/publisher_info_database.h]: done

File: components/brave_rewards/browser/database/publisher_info_database.cc (hunk 3 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/publisher_info_database.cc]: done

File: components/brave_rewards/browser/database/publisher_info_database.cc (hunk 4 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/publisher_info_database.cc]: not part of the PR, but fixed

File: components/brave_rewards/browser/database/database_media_publisher_info.h (hunk 5 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/database_media_publisher_info.h]: done

File: components/brave_rewards/browser/database/database_media_publisher_info.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/database_media_publisher_info.cc]: https://github.com/brave/brave-browser/issues/7145

**Pull request merged**